Gaming The Syllabus/Assets/Scripts/EventBasedInteractable.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+[RequireComponent(typeof(Collider2D))]
+public class EventBasedInteractable : MonoBehaviour, IInteractable
+{
+    [Tooltip("Higher values are interacted with first.")]
+    [SerializeField] private int interactionPriority;
+    [SerializeField] private UnityEvent onInteract;
+
+    public int InteractionPriority { get => interactionPriority; }
+
+    public void Interact()
+    {
+        onInteract?.Invoke();
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.CompareTag("Player"))
+        {
+            Player.AddInteractable(this);
+        }
+    }
+
+    private void OnTriggerExit2D(Collider2D collision)
+    {
+        if (collision.CompareTag("Player"))
+        {
+            Player.RemoveInteractable(this);
+        }
+    }
```

Comment:
I would have gone about this most likely in a more complex and complicated way. Glad you were able to accomplish it much more simply lol

Reply:
I have done this probably 50 different ways on 50 different projects and hated every single way I've done it except this one lol... have spent way too much time fighting with this sort of stuff